lib/acyclicGraph.ml

```diff
@@ -359,13 +359,16 @@ module Make (Point:Point) = struct
     let ltle =
       let fold acc n =
         let fold u strict acc =
-          if strict then PMap.add u strict acc
-          else if PMap.mem u acc then acc
-          else PMap.add u false acc
+          match PMap.find u acc with
+          | true -> acc
+          | false -> if strict then PMap.add u true acc else acc
+          | exception Not_found -> PMap.add u strict acc
```

Comment:
How about
```suggestion
          PMap.update u
            (function
              | None -> Some strict
              | Some strict' -> Some (strict || strict'))
            acc
```
?

Reply:
Your variant allocates more. I don't know whether it's observable in practice but I'd prefer to stick to an allocation-free code when possible.